Makefile

```diff
@@ -863,6 +838,33 @@ ROCKSDB_MAJOR = $(shell egrep "ROCKSDB_MAJOR.[0-9]" include/rocksdb/version.h |
 ROCKSDB_MINOR = $(shell egrep "ROCKSDB_MINOR.[0-9]" include/rocksdb/version.h | cut -d ' ' -f 3)
 ROCKSDB_PATCH = $(shell egrep "ROCKSDB_PATCH.[0-9]" include/rocksdb/version.h | cut -d ' ' -f 3)
 
+# If NO_UPDATE_BUILD_VERSION is set we don't update util/build_version.cc, but
+# the file needs to already exist or else the build will fail
+ifndef NO_UPDATE_BUILD_VERSION
+git_date := $(shell date "+%Y-%m-%d %T")
+ifdef FORCE_GIT_SHA
+	git_sha := $(FORCE_GIT_SHA)
+else
+	git_sha := $(shell git rev-parse HEAD 2>/dev/null)
+	git_tag  := $(shell git symbolic-ref -q --short HEAD || git describe --tags --exact-match 2>/dev/null)
+	git_mod  := $(shell git status --short | grep -v "??" | wc -l 2>/dev/null)
+	ifeq ("$(git_mod)","           0")
+	  git_date := $(shell git log -1 --date=format:"%Y-%m-%d %T" --format="%ad" 2>/dev/null)
```

Comment:
And a comment here like

    # Use commit data+time instead of current date+time if no changes

Reply:
Done.